Sources/Customization/DefaultEventDispatcher.swift

```diff
@@ -241,43 +235,45 @@ open class DefaultEventDispatcher: BackgroundingCallbacks, OPTEventDispatcher {
     }
     
     func stopTimer() {
-        timer.performAtomic { (timer) in
+        timer.performAtomic { timer in
             timer.invalidate()
         }
         timer.property = nil
     }
+    
+    // MARK: - Tests
+
+    open func close() {
+        self.flushEvents()
+        self.lock.sync {}
+    }
+    
 }
 
 // MARK: - Notification Observers
 
 extension DefaultEventDispatcher {
     
     func addProjectChangeNotificationObservers() {
-        observerProjectId = NotificationCenter.default.addObserver(forName: .didReceiveOptimizelyProjectIdChange, object: nil, queue: nil) { [weak self] (_) in
-            self?.logger.d("Event flush triggered by datafile projectId change")
-            self?.flushEvents()
-        }
+        observers.append(
```

Comment:
in which case projectId can be changed?

Reply:
It can happen with multi-clients. Since event queue is shared, it's a good time to flush events when a new project datafile is downloaded.